### PR TITLE
Fixes roleauthority landmark iteration

### PR DESCRIPTION
**Pull request by spookydonut**

### CODE REVIEW

File: code/game/jobs/role_authority.dm [hunk 1]
```diff
@@ -336,11 +336,8 @@ sorts them out by their department.
 	if(late_join) //If they late joined, we put them in cryo.
 		M.loc = late_join
 	else
-		var/i
-		var/obj/effect/landmark/L //To iterate.
 		var/obj/effect/landmark/S //Starting mark.
-		for(i in landmarks_list)
-			L = i
+		for(var/obj/effect/landmark/L in landmarks_list) // This whole thing is fucking awful
```
Comment (ninjanomnom) [inline code/game/jobs/role_authority.dm]: This should stay a typeless loop since the landmarks_list shouldn't contain anything else

Comment (spookydonut) [reply code/game/jobs/role_authority.dm]: This is just standardisation since all other uses of landmarks_list are type checked the same way

Comment (ninjanomnom) [reply code/game/jobs/role_authority.dm]: I don't care what the others are doing, this is incorrect

File: code/game/jobs/role_authority.dm [hunk 2]
```diff
@@ -336,10 +336,12 @@ sorts them out by their department.
 	if(late_join) //If they late joined, we put them in cryo.
 		M.loc = late_join
 	else
-		var/i
 		var/obj/effect/landmark/L //To iterate.
 		var/obj/effect/landmark/S //Starting mark.
-		for(i in landmarks_list)
+		for(var/i in landmarks_list) // This whole thing is fucking awful
+			if(!i)
+				stack_trace(i)
```
Comment (ninjanomnom) [inline code/game/jobs/role_authority.dm]: This takes a text arg for the error message, add a real message here

**Pull request merged**